Docs: Add a tutorial on how to create a block-based theme

carolinan opened the pull request:
##  Description

Adds a tutorial on how to create a block-based theme.
Closes #23698


-Image links will need to be updated.

carolinan committed:
Merge pull request #1 from WordPress/master

sync

carolinan committed:
Merge pull request #3 from WordPress/master

sync

carolinan committed:
Merge pull request #5 from WordPress/master

sync

carolinan committed:
Merge pull request #6 from WordPress/master

sync with master

carolinan committed:
Merge pull request #8 from WordPress/master

sync with master

carolinan committed:
Add block based themes to the tutorial section

carolinan committed:
Update README.md

carolinan committed:
Update README.md

carolinan committed:
Update README.md

carolinan committed:
Update README.md

carolinan committed:
Create adding-blocks.md

carolinan committed:
Update README.md

carolinan committed:
Update adding-blocks.md

carolinan committed:
Update adding-blocks.md

carolinan committed:
Update adding-blocks.md

carolinan committed:
Update adding-blocks.md

carolinan committed:
Update adding-blocks.md

carolinan committed:
Update README.md

carolinan committed:
Update README.md

carolinan committed:
Update adding-blocks.md

carolinan committed:
Update README.md

carolinan committed:
Add files via upload

carolinan committed:
Add files via upload

carolinan committed:
Add files via upload

carolinan committed:
Add files via upload

carolinan committed:
Update adding-blocks.md

carolinan committed:
Update adding-blocks.md

carolinan committed:
Update adding-blocks.md

carolinan committed:
Update adding-blocks.md

carolinan committed:
Update adding-blocks.md

gziolo reviewed (commented):
Thank you for starting this PR 👍

Can we host all images on wordpress.org? Git isn’t the best place to store media because it quickly increases the size of repository for all contributors. It keeps all copies that were ever included in its history.

carolinan commented:
I am not sure how :) Do we open a meta ticket once the content has been reviewed?

carolinan committed:
Update README.md

Fixed a typo.

kjellr commented on docs/designers-developers/developers/tutorials/block-based-themes/README.md:
```diff
@@ -0,0 +1,375 @@
+Templates can optionally include structural template parts, for example a header, footer or sidebar.
+
+Each template or template part contains the [block grammar](https://developer.wordpress.org/block-editor/principles/key-concepts/#blocks), the HTML, for the selected blocks.
+The block HTML is generated in and exported from the **site editor**. It can also be added to the themes HTML files manually.
```
```suggestion
The block HTML is generated in and exported from the **site editor**. It can also be added to the theme's HTML files manually.
```

Minor — missing an apostrophe here.

kjellr commented on docs/designers-developers/developers/tutorials/block-based-themes/README.md:
```diff
@@ -0,0 +1,375 @@
+The block HTML is generated in and exported from the **site editor**. It can also be added to the themes HTML files manually.
+
+### Required files and file structure
+A block based theme requires an index.php file, and index template file, a style.css file and a functions.php file.
```
```suggestion
A block based theme requires an index.php file, an index template file, a style.css file and a functions.php file.
```
Small typo.

kjellr commented on docs/designers-developers/developers/tutorials/block-based-themes/README.md:
```diff
@@ -0,0 +1,375 @@
+endif; // myfirsttheme_setup
+add_action( 'after_setup_theme', 'myfirsttheme_setup' );
+
+function myfirsttheme_scripts() {
```
```suggestion
/**
 * Enqueue theme scripts and styles.
 */
function myfirsttheme_scripts() {
```
Adding a quick comment to explain this section.

kjellr commented on docs/designers-developers/developers/tutorials/block-based-themes/README.md:
```diff
@@ -0,0 +1,375 @@
+Create an index.php file.
+This file is used as a fallback if the theme is activated when full site editing is not enabled. 
+You may leave the file empty for this tutorial.
+
```
Before moving on to the next section, would it be helpful to include a quick overview of what the file structure should look like at this point? 

```
 theme
 |__ style.css
 |__ functions.php
 |__ index.php
 |__ block-templates
 	|__ (empty folder)
 |__ block-template-parts
 	|__ (empty folder)
 ```

kjellr commented on docs/designers-developers/developers/tutorials/block-based-themes/adding-blocks.md:
```diff
@@ -0,0 +1,65 @@
+# Adding blocks to you theme
```
```suggestion
# Adding blocks to your theme
```

Minor typo.

kjellr commented on docs/designers-developers/developers/tutorials/block-based-themes/README.md:
````diff
@@ -0,0 +1,375 @@
+		}
+	}
+}
+```
````
To close off this part of the tutorial, I think it would be helpful to point people to the Adding Blocks page.

gziolo commented:
> I am not sure how :) Do we open a meta ticket once the content has been reviewed?

Good question. I never did that, but I believe @mapk should know how the process looks like.

kjellr reviewed (commented):
This is looking great! Thanks for writing this out, @carolinan! I've uploaded the images to `wordpress.org/gutenberg/`, which is where we've hosted other documentation images in the past: 

```
https://wordpress.org/gutenberg/files/2020/07/block-based-themes-template-menu.png
https://wordpress.org/gutenberg/files/2020/07/block-based-themes-save.png
https://wordpress.org/gutenberg/files/2020/07/block-based-themes-page-menu.png
https://wordpress.org/gutenberg/files/2020/07/block-based-themes-page-menu.png
https://wordpress.org/gutenberg/files/2020/07/block-based-themes-editor-template-part.png
https://wordpress.org/gutenberg/files/2020/07/block-based-themes-appearance-template-parts.png
```
So feel free to swap those out. 

You'll need to also run the documentation scripts since this adds new pages — I can't find the documentation for those (oddly enough 😂), but maybe someone else can point you in the right direction.

gziolo commented:
> You'll need to also run the documentation scripts since this adds new pages — I can't find the documentation for those (oddly enough 😂), but maybe someone else can point you in the right direction.

It's here:
https://github.com/WordPress/gutenberg/blob/master/docs/contributors/document.md#create-a-new-document

carolinan committed:
Apply suggestions from code review

Co-authored-by: Kjell Reigstad <kjell@kjellr.com>

carolinan committed:
Merge pull request #15 from WordPress/master

keeping branch up to date

carolinan committed:
Deleted images and replaced the image links.

Deleted images and replaced the image links.
Fixed typos.
Renamed the adding-blocks file to include the tutorial topic.
Updated toc.json and ran docs:build.

carolinan commented:
Got feedback that it is confusing how to add the post loop

kjellr commented:
> Got feedback that it is confusing how to add the post loop

Maybe a screenshot would help there?

mkaz commented on docs/designers-developers/developers/tutorials/block-based-themes/README.md:
```diff
@@ -16,7 +16,7 @@ Additional documentation for block-based themes is available at https://develope
  3. [Creating the templates and template parts](/docs/designers-developers/developers/tutorials/block-based-themes/README.md#creating-the-templates-and-template-parts)
  4. [Experimental-theme.json -Global styles](/docs/designers-developers/developers/tutorials/block-based-themes/README.md##experimental-themejson--global-styles)
- 5. [Adding blocks](/docs/designers-developers/developers/tutorials/block-based-themes/adding-blocks.md)
+ 5. [Adding blocks](docs/designers-developers/developers/tutorials/block-based-themes/block-based-themes-2-adding-blocks.md)
```
You do want the slash at the beginning, see [the Using Links in docs guide](https://developer.wordpress.org/block-editor/contributors/document/#using-links)

```suggestion
 5. [Adding blocks](/docs/designers-developers/developers/tutorials/block-based-themes/block-based-themes-2-adding-blocks.md)
```

mkaz commented on docs/designers-developers/developers/tutorials/block-based-themes/README.md:
```diff
@@ -32,7 +32,7 @@ Each template or template part contains the [block grammar](https://developer.wo
 
 ### Required files and file structure
-A block based theme requires an index.php file, and index template file, a style.css file and a functions.php file.
+A block based theme requires an index.php file, an index template file, a style.css file and a functions.php file.
```
Oxford comma 😉

```suggestion
A block based theme requires an index.php file, an index template file, a style.css file, and a functions.php file.
```

mkaz reviewed (changes requested):
Looks good, a link fix and a minor punctuation tweak.

carolinan committed:
Update docs/designers-developers/developers/tutorials/block-based-themes/README.md

Co-authored-by: Marcus Kazmierczak <marcus@mkaz.com>

carolinan committed:
Update docs/designers-developers/developers/tutorials/block-based-themes/README.md

Co-authored-by: Marcus Kazmierczak <marcus@mkaz.com>

kjellr commented:
This is looking great, thank you @carolinan! I'll merge this and we can continue iterating in new PRs as needed. 🙌
